lib/ansible/modules/packaging/os/satellite_subscription.py

```diff
@@ -0,0 +1,442 @@
+  unique:
+    description:
+     - Whether to remove all subscriptions to this host.
+     - When set to C(yes) remove all other subscriptions attached to this host.
+    type: bool
+    default: no
+  verify_ssl:
+    description:
+      - Check ssl certificates?
+    type: bool
+    default: True
+"""
+
+EXAMPLES = """
+- name: Assign subscriptions, remove all existing subscriptions (run against localhost)
+  satellite_subscription:
+    satellite_hostname: "{{ satellite_fqdn }}"
+    username: "{{ satellite_admin_user }}"
+    password: "{{ satellite_admin_pass }}"
+    state: present
+    content_host: "{{ groups.rhel_servers }}"
+    subscription:
+      - product_id=RH000XX
+      - name=EPEL
+      - name=Extra
+    unique: yes
+    organization: 1
+  delegate_to: localhost
+
+- name: Assign subscriptions rhel servers, remove all existing subscriptions (run against inventory hosts/group)
+  satellite_subscription:
+    satellite_hostname: "{{ satellite_fqdn }}"
+    username: "{{ satellite_admin_user }}"
+    password: "{{ satellite_admin_pass }}"
+    state: present
+    content_host:
+      - "{{ inventory_hostname }}"
+    subscription: "{{ redhat_subscriptions }}"
+    unique: true
+    organization: 1
+  delegate_to: localhost
+
+- name: remove a subscription from hosts
+  satellite_subscription:
+    satellite_hostname: "{{ satellite_fqdn }}"
+    username: "{{ satellite_admin_user }}"
+    password: "{{ satellite_admin_pass }}"
+    state: absent
+    content_host:
+      - server1
+      - server2
+    subscription:
+      - "Red Hat Enterprise Linux Server Entry Level, Self-support"
+    organization: 1
+
+- name: run auto-attach for a host
+  satellite_subscription:
+    satellite_hostname: "{{ satellite_fqdn }}"
+    username: "{{ satellite_admin_user }}"
+    password: "{{ satellite_admin_pass }}"
+    state: autoattach
+    content_host: server1
+    organization: 1
+
+- name: assign VDC guest subscriptions to Satellite content hosts
+  satellite_subscription:
+    satellite_hostname: "{{ satellite_fqdn }}"
+    username: "{{ satellite_admin_user }}"
+    password: "{{ satellite_admin_pass }}"
+    state: vdcguests
+    content_host:
+      - "{{ inventory_hostname }}"
+    organization: "{{ satellite_organization_id }}"
+    verify_ssl: "{{ satellite_verify_ssl }}"
+  delegate_to: localhost
+
+- name: Optimize by replacing multiple entitlements from subscription_id 242 with the new subscription (no changes on hosts consuming a single entitlement)
+  satellite_subscription:
+    satellite_hostname: "{{ satellite_fqdn }}"
+    username: "{{ satellite_admin_user }}"
+    password: "{{ satellite_admin_pass }}"
+    state: optimize
+    content_host:
+      - "subscription_id=242"
+    subscription:
+      - "Red Hat Enterprise Linux Server Entry Level, Self-support"
+    organization: "{{ satellite_organization_id }}"
+    verify_ssl: "{{ satellite_verify_ssl }}"
+  delegate_to: localhost
+"""
+
+RETURN = """
+stdout:
```

Comment:
I would call this `msg` and not `stdout`. We tend to use the latter for command output.